test/test_model.py

```diff
@@ -3337,6 +3337,11 @@ def test_open_port_error(self):
             ['open-port', '8080/ftp'],
         ])
 
+        with self.assertRaises(ops.ModelError) as cm:
+            self.unit.open_port('icmp', 8000)
+        with self.assertRaises(ops.ModelError) as cm:
+            self.unit.open_port('udp')
+
```

Comment:
We can probably drop these tests too once we've dropped the extra logic in `open_port`. We don't need to test the hook tool error handling itself, and we've already tested above that if it returns an error we raise `ModelError`.